apps/barista-design-system/src/shared/services/scroll-spy.service.ts

```diff
@@ -14,212 +14,120 @@
   }
 
-  private _getContentHeight(): number {
-    return this._document.body.scrollHeight || Number.MAX_SAFE_INTEGER;
+  /**
+   * Evaluates by comparing the users scroll position with the element top property and return the id of an element
+   * that should be highlighted
+   */
+  findActiveItem(): string {
+    // The element id of the item to be highlighted
+    let activeItem: string | null = null;
+    this._zone.runOutsideAngular(() => {
+      const scrollTop = this._getScrollTop();
+      const contentHeight = this._getContentHeight();
+      const viewportHeight = this._getViewportHeight();
+      for (let i = 0; i < this.headlineElements.length; i++) {
+        // Check whether an elements top value is smaller then the users scroll top position and offset
+        // resulting in setting that value as `active`
+        if (this.headlineElements[i].top <= scrollTop + this.OFFSET) {
+          activeItem = this.headlineElements[i].element.id;
+        } else if (scrollTop + viewportHeight >= contentHeight) {
+          // Special case when user is at the bottom of the page and there's no way tthe last item will be active
+          activeItem = this.headlineElements[this.headlineElements.length - 1]
+            .element.id;
+        }
+      }
+    });
+    return activeItem!;
   }
 
+  /** Current position of user in Browser */
   private _getScrollTop(): number {
     return (this._platform.isBrowser && window.pageYOffset) || 0;
   }
 
-  private _getViewportHeight(): number {
-    return (this._platform.isBrowser && window.innerHeight) || 0;
-  }
-
-  /*
-   * The size of the window has changed. Re-calculate all affected values,
-   * so that active elements can be determined efficiently on scroll.
-   */
-  private _onResize(): void {
-    const contentHeight = this._getContentHeight();
-    const viewportHeight = this._getViewportHeight();
-    const scrollTop = this._getScrollTop();
-
-    this._lastContentHeight = contentHeight;
-    this._lastMaxScrollTop = contentHeight - viewportHeight;
-
-    for (const group of this._spiedElementGroups) {
-      group.calibrate(scrollTop, this._topOffset);
-    }
-  }
-
-  /*
-   * Determine which element for each ScrollSpiedElementGroup is active. If the content height has
-   * changed since last check, re-calculate all affected values first.
-   */
-  private _onScroll(): void {
-    if (this._lastContentHeight !== this._getContentHeight()) {
-      // Something has caused the scroll height to change.
-      // (E.g. image downloaded, accordion expanded/collapsed etc.)
-      this._onResize();
-    }
-
-    const scrollTop = this._getScrollTop();
-    const maxScrollTop = this._lastMaxScrollTop;
-
-    for (const group of this._spiedElementGroups) {
-      group.onScroll(scrollTop, maxScrollTop);
-    }
+  /** Height of the whole content */
+  private _getContentHeight(): number {
+    return this._document.body.scrollHeight || Number.MAX_SAFE_INTEGER;
```

Comment:
Why do we use the max integer here? I think no document is that long. Please explain the intention here.

Reply:
It's not needed. Leftover from the old version and I overlooked it